lib/test/revision.go

```diff
@@ -139,3 +139,17 @@ func RevisionListWithService(r *KnRunResultCollector, serviceNames ...string) {
 		}
 	}
 }
+
+// RevisionDeleteWithPruneOption verifies removeing all unreferenced revisions for a given service in sync mode
+func RevisionDeleteWithPruneOption(r *KnRunResultCollector, serviceName string) {
+	out := r.KnTest().Kn().Run("revision", "delete", "--prune", serviceName)
+	assert.Check(r.T(), util.ContainsAll(out.Stdout, "Revision", "deleted", "namespace", r.KnTest().Kn().Namespace()))
+	r.AssertNoError(out)
+}
+
+// RevisionDeleteWithPruneAllOption verifies removeing all unreferenced revision in sync mode
+func RevisionDeleteWithPruneAllOption(r *KnRunResultCollector) {
+	out := r.KnTest().Kn().Run("revision", "delete", "--prune-all")
+	assert.Check(r.T(), util.ContainsAll(out.Stdout, "Revision", "deleted", "namespace", r.KnTest().Kn().Namespace()))
+	r.AssertNoError(out)
```

Comment:
Could we add a verification step here that asserts revision list count at least? To check expected number of revision is present or not. It can be either here or a separate `Verify...` function, as the same check is valid for the above `func` as well.